ros2lifecycle/ros2lifecycle/api/__init__.py

```diff
@@ -27,14 +27,14 @@ def get_node_names(*, node, include_hidden_nodes=False):
         node=node, include_hidden_nodes=include_hidden_nodes)
     service_names_and_types = get_service_names_and_types(node=node)
     return [
-        n for n in node_names
-        if _has_lifecycle(n, service_names_and_types)]
+        n.full_name for n in node_names
```

Comment:
Would it be more consistent with the same function make in `ros2node` to return the "full" `n` here and let the callers pick the`full_name` from it?

Reply:
I can change it if you want. But given that this function anyway only really serves for the lifecycle implementation i didn't do it at first.